### DISCUSSION TITLE
Has anyone tried AMP BF16 pretraining from scratch?

### DISCUSSION THREAD

**Discussion by kyleliang919** [mlfoundations/open_clip#271] [Q&A]
On YFCC, I am getting consistently worse zero-shot results from FP32/FP16 pertaining on the same set of hyperparameters, despite the training loss roughly matches. Just wondering if bf16 is particularly bad for models like CLIP.

**Comment by rom1504** [comment]
We trained many models with amp bf16 and didn't see a difference from amp
fp16.

What model are you training specifically?

We did fairly large scale (10k to 100k GPU hours) so that might be a
difference to your setup?

**Reply by kyleliang919** [reply]
CLIP ViT-B/32 with 32 way DP

**Reply by kyleliang919** [reply]
I am trying to replicate the YFCC experiment in the paper following the exact hyperparameters.

**Comment by rom1504** [comment]
Ok that's pretty small scale. Maybe bf16 causes issues only at this scale.
Anyway stability issues appear only at larger scale, so you can use fp16
for sure
